### PR TITLE
FIX error of load order when using configure in preview|config.js

### PR DESCRIPTION
**Pull request by ndelangen**
see:
https://app.circleci.com/pipelines/github/storybookjs/storybook/7850/workflows/86d66710-5483-4d98-b590-c76aea8aec6a/jobs/203561

solution is related to: https://github.com/storybookjs/storybook/issues/9819?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDcxMDg4MjI1NjozMDcwMzg5#issuecomment-586764534

### CODE REVIEW

**Comment by tmeasday** [comment]
Can you explain the idea here a bit more?

**Comment by ndelangen** [comment]
preview & config can call `configure` which should always happen AFTER all global decorators & args are set.

File: scripts/build-storybooks.js [hunk 1]
```diff
@@ -148,7 +148,9 @@ const handleExamples = async deployables => {
     const out = p(['built-storybooks', d]);
     const cwd = p(['examples', d]);
 
-    await exec(`yarn`, [`build-storybook`, `--output-dir=${out}`, '--quiet'], { cwd });
+    await exec(`yarn`, [`build-storybook`, `--output-dir=${out}`, '--quiet', '--debug-webpack'], {
```
Comment (shilman) [inline scripts/build-storybooks.js]: is the `--debug-webpack` necessary? i'd think it would generate a lot of noise

Comment (ndelangen) [reply scripts/build-storybooks.js]: yeah we can revert that, used it for debugging

**Review by shilman — COMMENTED** [review]
This is pretty gnarly

File: lib/core/src/server/preview/entries.js [hunk 2]
```diff
@@ -1,5 +1,34 @@
 import path from 'path';
 import { logger } from '@storybook/node-logger';
+import stable from 'stable';
+import { loadPreviewOrConfigFile } from '../utils/load-preview-or-config-file';
+
+export const sortEntries = entries => {
+  const isGenerated = /generated-(config|other)-entry/;
+  const isGeneratedConfig = /(?:preview|config)\..+-generated-config-entry/;
+
+  return stable(entries.slice(0), (a, b) => {
```
Comment (shilman) [inline lib/core/src/server/preview/entries.js]: "preview & config can call configure which should always happen AFTER all global decorators & args are set." - better explanation

**Review by shilman — APPROVED** [review]
i'm ok with this with better comment & cleanup the webpack. @tmeasday?

**Comment by tmeasday** [comment]
Ok yeah it makes sense to me. Let’s make sure in the code the reason for it is clear

**Comment by ndelangen** [comment]
Is this code comment acceptable?

**Comment by tmeasday** [comment]
I think we need a bit more explanation. Something like:

> We need to ensure that all parameters and decorators that are added by preview entrypoints added by addons happen *before* any `configure()` calls executed by the user's `preview.js` (or `config.js`), or by `main.js:stories`. As those addons will create generated entries, this means we need to ensure all generated entries come before all other entries (generated or otherwise).

**Comment by ndelangen** [comment]
fixed!

**Pull request merged**